### PR TITLE
Add test cases for automatically updating milestone due date after a successful release

### PR DESCRIPTION
**Pull request by tarunvijwani**
This PR adds the test cases for automatically updating milestone due date after a successful release. 

### Changes

- Add test cases for updating milestones.
- Add mock milestone data to the iterator in the test helper file. 
- Add the `updateMilestone` function in the test helper file. 

### Testing
- Check out this branch
- Run `npm run test update-milestone` command in the terminal. 
- Confirm all tests are passing.

### CODE REVIEW

File: lib/automations/update-milestone/test/update-milestone-handler.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const { gimmeOctokit, gimmeContext, gimmeApp } = require( '@testHelpers' );
+
+describe( 'Update milestone after successful release', () => {
+	let octokit, context, app;
+	beforeEach( () => {
+		octokit = gimmeOctokit();
+		context = gimmeContext( 'release', 'published' );
+		app = gimmeApp( 'update-milestone' );
+	} );
+	it( 'Does not updates the milestone if milestone not present', async () => {
```
Comment (opr) [inline lib/automations/update-milestone/test/update-milestone-handler.js]:
```suggestion
	it( 'Does not update the milestone if the milestone is not present', async () => {
```

File: lib/automations/update-milestone/test/update-milestone-handler.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		await app( context, octokit, {
+			targetMilestone: '6.0.0',
+		} );
+		expect(
+			octokit.issues.listMilestones.endpoint.merge
+		).toHaveBeenCalled();
+		expect( octokit.issues.updateMilestone ).not.toHaveBeenCalled();
+	} );
+	it( 'Does not updates the milestone if milestone have a due date', async () => {
```
Comment (opr) [inline lib/automations/update-milestone/test/update-milestone-handler.js]:
```suggestion
	it( 'Does not update the milestone if the milestone has a due date', async () => {
```

**Review by opr — APPROVED** [review]
Code looks OK, just left some suggestions about naming of the tests, approving because these are non-blocking.

I am curious though, why don't we close the milestone if it has a due date?

**Comment by tarunvijwani** [comment]
> Code looks OK, just left some suggestions about naming of the tests, approving because these are non-blocking.
> 
> I am curious though, why don't we close the milestone if it has a due date?

@opr We are already closing the milestone in the [previous step when the release branch ](https://github.com/woocommerce/automations/pull/54)is created. In this step, we only update the due date after the successful release.

**Pull request merged**